rectools/visuals/metrics_app.py

```diff
@@ -0,0 +1,120 @@
+        show_legend: bool = True,
+        auto_display: bool = True,
+        plotly_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+    ) -> "MetricsApp":
+        """TODO: add docstring"""
+        return cls(df_metrics_data, show_legend, auto_display, plotly_kwargs)
+
+    def _validate_input_data(self) -> None:
+        if not isinstance(self.metrics_data, pd.DataFrame):
+            raise ValueError("Incorrect input type. `metrics_data` should be a DataFrame")
+        if Columns.Split not in self.metrics_data.columns:
+            raise KeyError("Missing 'Split' column in `metrics_data` DataFrame")
+        if Columns.Model not in self.metrics_data.columns:
+            raise KeyError("Missing 'Model' column in `metrics_data` DataFrame")
+        if len(self.metrics_data.columns) < 3:
+            raise KeyError("`metrics_data` DataFrame assumed to have at least one metric column")
+
+    @property
+    def metric_names(self) -> tp.List[str]:
+        """TODO: add docstring"""
+        non_metric_columns = {Columns.Split, Columns.Model}
+        return [col for col in self.metrics_data.columns if col not in non_metric_columns]
+
+    @property
+    def n_folds(self) -> int:
+        """TODO: add docstring"""
+        return self.metrics_data[Columns.Split].nunique()
+
+    def _make_chart_data(self, fold_number: int) -> pd.DataFrame:
+        return self.metrics_data[self.metrics_data[Columns.Split] == fold_number].drop(columns=Columns.Split)
+
+    def _make_chart_data_avg(self) -> pd.DataFrame:
+        return self.metrics_data.drop(columns=Columns.Split).groupby(Columns.Model, sort=False).mean().reset_index()
+
+    def _update_chart(
+        self,
+        metric_x: widgets.Dropdown,
+        metric_y: widgets.Dropdown,
+        use_avg: widgets.Checkbox,
+        fold: widgets.Dropdown,
+        scatter_chart: widgets.Output,
+    ) -> None:
+        with scatter_chart:
+            scatter_chart.clear_output(wait=True)
+            chart_data = self._make_chart_data_avg() if use_avg.value else self._make_chart_data(fold.value)
+            self.fig = px.scatter(
+                chart_data, x=metric_x.value, y=metric_y.value, color=Columns.Model, **self.plotly_kwargs
+            )
+            self.fig.layout.update(showlegend=self.show_legend)
+            self.fig.show()
+
+    def _toggle_fold_number_visibility(self, fold: widgets.Dropdown, use_avg: widgets.Checkbox) -> None:
+        fold.layout.visibility = "hidden" if use_avg.value else "visible"
+
+    def display(self) -> None:
+        """TODO: add dosctring"""
+        metric_x = widgets.Dropdown(description="Metric X:", options=self.metric_names, value=self.metric_names[0])
+        metric_y = widgets.Dropdown(description="Metric Y:", options=self.metric_names, value=self.metric_names[-1])
+        use_avg = widgets.Checkbox(description="Avg folds", value=True)
+        fold = widgets.Dropdown(description="Fold number:", options=list(range(self.n_folds)), value=0)
```

Comment:
Default value (as `value=` here) should also be taken from the `Columns.Split` first value from sorted list because users can provide metrics data calculated outside RecTools

Reply:
Sure, done